### ISSUE TITLE
clientv3 can't be compiled with latest gRPC change

### ISSUE THREAD

**Issue by gyuho** [coreos/etcd#5561]
```
# github.com/coreos/dbtester/vendor/github.com/coreos/etcd/clientv3
```
> vendor/github.com/coreos/etcd/clientv3/client.go:309: oldconn.State undefined (type *grpc.ClientConn has no field or method State)
vendor/github.com/coreos/etcd/clientv3/client.go:311: oldconn.WaitForStateChange undefined (type *grpc.ClientConn has no field or method WaitForStateChange)

Currently we don't force users to use our gRPC dependencies, so this will happen.

I am creating this issue so we don't forget.

**Comment by heyitsanthony** [comment]
For the sake of completeness, I'll note the reason the code does that is so grpc won't queue the Close and keep some reconnection goroutine running (making the leak detector complain). Hopefully removing it won't cause any headaches...

**Comment by mkumatag** [comment]
I've also faced this issue while getting v3client
```
[root@jupiter-vm1164 src]# go get github.com/coreos/etcd/clientv3
# github.com/coreos/etcd/clientv3
github.com/coreos/etcd/clientv3/client.go:309: oldconn.State undefined (type *grpc.ClientConn has no field or method State)
github.com/coreos/etcd/clientv3/client.go:311: oldconn.WaitForStateChange undefined (type *grpc.ClientConn has no field or method WaitForStateChange)
[root@jupiter-vm1164 src]# 
```

Do we have any workaround for this?

**Comment by mkumatag** [comment]
part of comit - https://github.com/grpc/grpc-go/commit/9c2d8546bf9fdbae6679bf11f1e420730277659a#diff-e1550a73f5d25064c8b586ec68d81a64 they have created new structure called addrConn and moved State to getState and WaitForStateChange to waitForStateChange. So we might to change the logic in the client file.

**Comment by heyitsanthony** [comment]
Fixed by #5583
